ls1intum/Artemis
# [Exam] Programming submissions in summary page

### Checklist

### Motivation and Context


Programming exercise submissions and their commit hashes were not correctly displayed in the exam-summary component. Unprepared programming exercises did show the wrong synchronization status in the navigation bar. Unprepared exercises had a wrong rendering behavior (were redrawn on changing to unprepared exercises)

### Description

- [x] Add  websocket subscription in exam-participation-component to get latest submission
- [x] Show commit hash in summary page if a commit was made
- [x] Navigation bar synchronization status correct when using programming exercises which were not prepared
- [x] Navigation bar status is correctly displayed after 'Refresh files' is executed
- [x] Changed check in exam-participation.component.html -> check for participation of exercise not activeExercise. This eliminates a bug when participating in an exam without initially prepared exercises. Generation of participations on first visit of an exercise lead to the destruction of other submission components.

### Steps for Testing


_Exam with prepared participations_
1. Create an exam with programming exercises with online editor, generate the StudentExams and prepare the exercises
2. Participate in the exam
3. Synchronization status of programming exercises should be correct (try 'Refresh Files')
4. When the exam ends, the correct commit hash should be displayed

_Exam without prepared participations_
1. Create an exam with programming exercises with online editor, generate the StudentExams and **do not** prepare the exercises
2. Participate in the exam
--> Participation for the exercise will be created when you navigate to the exercise the first time
3. Synchronization status of programming exercises should be correct (try 'Refresh Files')
4. When the exam ends, the correct commit hash should be displayed

### Screenshots

![image](https://user-images.githubusercontent.com/22955066/87569373-50866d00-c6c7-11ea-8551-fc4559248e96.png)
